libbeat/docs/command-reference.asciidoc

```diff
@@ -0,0 +1,583 @@
+only for testing {beatname_uc}.
+
+ifeval::["{beatname_lc}"=="packetbeat"]
+
+*`-O, --O`*::
+Read packets one by one by pressing _Enter_ after each. This option is useful
+only for testing {beatname_uc}.
+
+endif::[]
+
+*`--cpuprofile FILE`*::
+Writes CPU profile data to the specified file. This option is useful for
+troubleshooting {beatname_uc}.
+
+ifeval::["{beatname_lc}"=="packetbeat"]
+
+*`-devices`*::
+Prints the list of devices that are available for sniffing and then extis.
+
+endif::[]
+
+ifeval::["{beatname_lc}"=="packetbeat"]
+
+*`-dump FILE`*::
+Writes all captured packets to the specified file. This option is useful for
+troubleshooting {beatname_uc}.
+
+endif::[]
+
+*`-h, --help`*::
+Shows help for the `run` command.
+
+*`--httpprof [HOST]:PORT`*::
+Starts an http server for profiling. This option is useful for troubleshooting
+and profiling {beatname_uc}.
+
+ifeval::["{beatname_lc}"=="packetbeat"]
+
+*`-l N`*::
+Reads the pcap file `N` number of times. The default is 1. Use this option in
+combination with the `-I` option. For an infinite loop, use _0_. The `-l`
+option is useful only for testing {beatname_uc}.
+
+endif::[]
+
+*`--memprofile FILE`*::
+Writes memory profile data to the specified output file. This option is useful
+for troubleshooting {beatname_uc}.
+
+ifeval::["{beatname_lc}"=="filebeat"]
+
+//REVIEWERS: I've written this topic to be consitent to what I see when I run -help, but I'm wondering why we aren't also showing the options that begin with a dash. For example, in the modules doc we show: -modules=nginx,mysql,system. Is --modules preferred?
+
+*`--modules MODULE_LIST`*::
+Specifies a comma-separate list of modules to run. For example:
++
+["source","sh",subs="attributes"]
+-----
+{beatname_lc} run --modules nginx,mysql,system
+-----
++
+Rather than specifying the list of modules every time you run {beatname_uc},
+you can use the <<modules-command,`modules`>> command to enable and disable
+specific modules. Then when you run {beatname_uc}, it will run any modules
+that are enabled.
+
+//REVIEWERS: Is the above syntax correct for --modules? We used to say -modules=nginx,mysql,system. Not sure if it makes a difference, but I'm trying to be consistent with what users see when they view --help.
+
+endif::[]
+
+ifeval::["{beatname_lc}"=="filebeat"]
+
+*`--once`*::
+When the `--once` flag is used, {beatname_uc} starts all configured harvesters
+and prospectors, and runs each prospector until the harvesters are closed. If
+you set the `--once` flag, you should also set `close_eof` so the harvester is
+closed when the end of the file is reached. By default harvesters are closed
+after `close_inactive` is reached.
+
+
+//REVIEWERS: Is the behavior of the harvesters/prospectors described above still true?
+
+endif::[]
+
+//REVIEWERS: I'm not documenting --setup because it will be removed in 6.0. Are there any other flags that I need to remove? Or mention that they're being maintained for backwards compability?
```

Comment:
@exekias I think it might have been a warning message when I ran the command? I'll add it back to the docs.